docs/workflow.md

```diff
@@ -0,0 +1,149 @@
+<!--
+TODO: should we call "this project" Juno? I saw in communication.md we did,
+but I'm not sure
+ -->
+
+# Workflow
+
+One of the goals of this project is to help users adopt a more efficient
+workflow when editing Julia code. To understand how to gain this productivity
+boost, we will consider a common workflow for many Julia developers and a Juno
+based workflow.
+
+## Status quo
```

Comment:
Agreed. I was a bit on the fence about it myself.

I think maybe we could drop it here and put in a line like "see the worfklow tips section of the manual for a description of the current workflow for many Julia developers." What do you think?